src/Keyboard/index.tsx

```diff
@@ -0,0 +1,246 @@
+
+interface OnScreenKeyboardProps {
+  /**
+   * The text that's currently being searched
+   */
+  searchQuery: string;
+  /**
+   * A handler to update the search text
+   */
+  searchUpdateHandler: (arg0: string) => void;
+  /**
+   * A handler to initiate a search
+   */
+  triggerSearchHandler: () => void;
+}
+
+
+/**
+* Enum for handling the casing of the keyboard
+*/
+enum CASE {
+  /** lowercase layout */
+  DEFAULT,
+  /** next character uppercase, then back to lowercase */
+  SHIFTED,
+  /** uppercase layout */
+  LOCKED,
+  /** next character lowercase, then back to uppercase */
+  UNSHIFTED,
+}
+
+const OnScreenKeyboard = ({
+  searchQuery,
+  searchUpdateHandler,
+  triggerSearchHandler,
+}: OnScreenKeyboardProps) => {
+  /**
+   * Whether the keyboard should be shown on screen.
+   */
+  const [isVisible, setVisible] = useState(false);
+
+  /**
+   * Set a ref to the modal for updating placement, etc.
+   */
+  const keyboardModalRef = useRef<HTMLDivElement>(null);
+  const backdropRef = useRef<HTMLDivElement>(null);
+  /**
+   * The X,Y coordinates on the page at which the center point of the keyboard
+   * should be rendered
+   */
+  const [coordinates, setCoordinates] = useState<[number, number]>([540, 960]);
+
+  /**
+   * Handle setting the event listener to open the keyboard when clicking on
+   * the body, then close it when clicking outside the modal
+   */
+  useEffect(() => {
+    if (isVisible) {
+      const background = backdropRef.current;
+      const closeKeyboard = (evt: MouseEvent) => {
+        if (evt.target === background) {
+          searchUpdateHandler('')
+          setVisible(false);
+        };
+      }
+      if (background) {
+        background.addEventListener('click', closeKeyboard);
+      }
+      return(() => {
+        if (background) {
+           background.removeEventListener('click', closeKeyboard);
+        }
+      })
+    }
+    const openKeyboard = (evt: MouseEvent) => {
+      setVisible(true);
+      setCoordinates([evt.clientX, evt.clientY]);
+    };
+    document.body.addEventListener('click', openKeyboard);
+    return(() => {
+      document.body.removeEventListener('click', openKeyboard);
+    })
+  }, [
+    isVisible,
+    setVisible,
+    setCoordinates,
+    backdropRef,
+    searchUpdateHandler
+  ]);
+
+  /**
+  * Handles the Capitalization state of the keyboard
+  */
+  const [keyboardCase, setKeyboardCase] = useState<CASE>(CASE.DEFAULT);
+
+  /**
+   * Handles the logic for controlling the non-input keys, i.e shift, caps, and
+   * enter.
+   * The capitalization is not built into the keyboard library unfortunately,
+   * and I think that leaving out the upper/lower case functionality might be
+   * more confusing for users.
+   */
+
+  const otherKeyHandler = (button: string) => {
+    // Trigger a search on enter
+    if (button === '{enter}') {
+      setVisible(false);
+      searchUpdateHandler('');
+      triggerSearchHandler();
+    }
```

Comment:
I'm having trouble figuring out why exactly this is happening, but when I click on the enter button, it briefly closes the keyboard and then re-opens it (a little more to the right since the "enter" key is on the right of the keyboard). 

I added some `console.log` statements, and it looks like `openKeyboard` is getting called after clicking "enter." 
It seems like we need to stop the click from bubbling up to the event listener on the top level so that the keyboard does not reopen. I tried calling `event.stopPropogation` on the keyboard container div as a hacky workaround, but it didn't work:
![image](https://user-images.githubusercontent.com/29589689/129769692-dbc3c96c-d8d5-4c3f-86a1-f277d386f1a5.png)

Reply:
Recapping discussion on Slack: Neither of us can now reproduce this bug in Firefox or Chrome (and derivatives). Agreed that there's nothing we can really fix now, but we'll keep an eye out in case it comes back.